.github/workflows/lint.yml

```diff
@@ -86,6 +86,21 @@ jobs:
         run: |
           echo "::remove-matcher owner=mypy::"
 
+      - name: lint pyproject.toml
+        if: ${{ !cancelled() && (steps.deps.outcome == 'success') }}
+        run: |
+          make toml-lint
+
+      - name: format pyproject.toml
+        if: ${{ !cancelled() && (steps.deps.outcome == 'success') }}
+        run: |
+          make toml-fmt
+
+      - name: check for diff and fail if found
+        if: ${{ !cancelled() && (steps.deps.outcome == 'success') }}
+        run: |
+          git diff --exit-code || (echo "'pyproject.toml' formatting is incorrect. Please run ' make toml-fmt' and commit the changes." && exit 1)
```

Comment:
Make this git diff part of the previous step since the git diff is logically part of the check "error if formatting is not done".